### PR TITLE
[Enhancement] Refine profile to support visualization refactor(3)

### PR DESCRIPTION
**Pull request by liuyehcf**
This pr is an continuation work of #29063 and #30381, and the main objectives of this pr are:

1. Add metrics for partition topn operator
2. Change display name for topn and partition topn, instead of simpliy using`Sort`
3. Add `IsChild` is an operator is a child operator, like AggOperator is child operator of BucketOperator
4. Curb unnecessary logs
5. Use `TUnitType::NONE` to represent hierarchy.

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 14 / 16 (87.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/scheduler/QueryRuntimeProfile.java|1|2|50.00%|[516]|
|:large_blue_circle:|com/starrocks/common/util/ProfilingExecPlan.java|6|7|85.71%|[398]|
|:large_blue_circle:|com/starrocks/sql/ast/AnalyzeProfileStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/scheduler/dag/FragmentInstanceExecState.java|6|6|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 3 / 39 (07.69%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/bucket_process_operator.h|0|2|00.00%|[32, 63]|
|:large_blue_circle:|src/exec/pipeline/sort/local_partition_topn_sink.cpp|0|3|00.00%|[43, 44, 45]|
|:large_blue_circle:|src/util/runtime_profile.cpp|0|4|00.00%|[1001, 1002, 1003, 1009]|
|:large_blue_circle:|src/exec/pipeline/sort/local_partition_topn_context.cpp|0|1|00.00%|[88]|
|:large_blue_circle:|src/exec/partition/partition_hash_map.h|0|2|00.00%|[260, 350]|
|:large_blue_circle:|src/exec/partition/partition_hash_variant.cpp|0|17|00.00%|[135, 136, 137, 139, 140, 145, 146, 157, 158, 159, 161, 162, 166, 167, 168, 170, 171]|
|:large_blue_circle:|src/exec/pipeline/sort/local_partition_topn_context.h|0|2|00.00%|[71, 73]|
|:large_blue_circle:|src/exec/partition/chunks_partitioner.h|0|2|00.00%|[78, 224]|
|:large_blue_circle:|src/exec/pipeline/operator.cpp|3|6|50.00%|[58, 82, 83]|

File: be/src/connector/lake_connector.cpp [hunk 1]
```diff
@@ -693,7 +702,6 @@ void LakeDataSource::update_counter() {
     COUNTER_UPDATE(_io_ns_local_disk_timer, _reader->stats().io_ns_local_disk);
     COUNTER_UPDATE(_io_ns_remote_timer, _reader->stats().io_ns_remote);
     COUNTER_UPDATE(_io_ns_total_timer, _reader->stats().io_ns);
-    COUNTER_UPDATE(_io_statistics_timer, _reader->stats().io_ns);
```
Comment (fzhedu) [inline be/src/connector/lake_connector.cpp:696]: don't update `_io_statistics`

File: be/src/exec/pipeline/sort/local_partition_topn_context.h [hunk 2]
```diff
@@ -68,6 +68,10 @@ class LocalPartitionTopnContext {
     // Pull one chunk from sorters or passthrough_buffer
     StatusOr<ChunkPtr> pull_one_chunk();
 
+    bool is_passthrough() const { return _chunks_partitioner->is_passthrough(); }
+
+    size_t num_partitions() const { return _partition_num; }
```
Comment (fzhedu) [inline be/src/exec/pipeline/sort/local_partition_topn_context.h:73]: `_partition_num` is already set before call this ?

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 26 / 28 (92.86%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/scheduler/QueryRuntimeProfile.java|1|2|50.00%|[516]|
|:large_blue_circle:|com/starrocks/common/util/ProfilingExecPlan.java|6|7|85.71%|[398]|
|:large_blue_circle:|com/starrocks/sql/ast/AnalyzeProfileStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/scheduler/dag/FragmentInstanceExecState.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/RuntimeProfileParser.java|11|11|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/RuntimeProfile.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 3 / 39 (07.69%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/bucket_process_operator.h|0|2|00.00%|[32, 63]|
|:large_blue_circle:|src/exec/pipeline/sort/local_partition_topn_sink.cpp|0|3|00.00%|[43, 44, 45]|
|:large_blue_circle:|src/util/runtime_profile.cpp|0|4|00.00%|[1001, 1002, 1003, 1009]|
|:large_blue_circle:|src/exec/pipeline/sort/local_partition_topn_context.cpp|0|1|00.00%|[88]|
|:large_blue_circle:|src/exec/partition/partition_hash_map.h|0|2|00.00%|[260, 350]|
|:large_blue_circle:|src/exec/partition/partition_hash_variant.cpp|0|17|00.00%|[135, 136, 137, 139, 140, 145, 146, 157, 158, 159, 161, 162, 166, 167, 168, 170, 171]|
|:large_blue_circle:|src/exec/pipeline/sort/local_partition_topn_context.h|0|2|00.00%|[71, 73]|
|:large_blue_circle:|src/exec/partition/chunks_partitioner.h|0|2|00.00%|[78, 224]|
|:large_blue_circle:|src/exec/pipeline/operator.cpp|3|6|50.00%|[58, 82, 83]|

File: be/src/exec/pipeline/sort/local_partition_topn_context.h (hunk 2 above)
Comment (liuyehcf) [reply be/src/exec/pipeline/sort/local_partition_topn_context.h:73]: Yes, I'll call it after it is set.

File: be/src/connector/lake_connector.cpp (hunk 1 above)
Comment (liuyehcf) [reply be/src/connector/lake_connector.cpp:696]: Didn't get what you mean. Could you be more specific?

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 4 / 92 (04.35%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/bucket_process_operator.h|0|2|00.00%|[32, 63]|
|:large_blue_circle:|src/exec/pipeline/sort/local_partition_topn_sink.cpp|0|3|00.00%|[43, 44, 45]|
|:large_blue_circle:|src/util/runtime_profile.cpp|0|4|00.00%|[1001, 1002, 1003, 1009]|
|:large_blue_circle:|src/connector/lake_connector.cpp|0|52|00.00%|[531, 532, 533, 534, 535, 537, 539, 540, 541, 542, 543, 544, 545, 546, 547, 548, 549, 552, 553, 554, 555, 556, 557, 558, 559, 560, 561, 562, 563, 564, 566, 571, 572, 574, 575, 577, 578, 579, 580, 581, 584, 586, 588, 590, 592, 593, 594, 595, 596, 598, 599, 600]|
|:large_blue_circle:|src/exec/pipeline/sort/local_partition_topn_context.cpp|0|1|00.00%|[88]|
|:large_blue_circle:|src/exec/partition/partition_hash_map.h|0|2|00.00%|[260, 350]|
|:large_blue_circle:|src/exec/partition/partition_hash_variant.cpp|0|17|00.00%|[135, 136, 137, 139, 140, 145, 146, 157, 158, 159, 161, 162, 166, 167, 168, 170, 171]|
|:large_blue_circle:|src/exec/pipeline/sort/local_partition_topn_context.h|0|2|00.00%|[71, 73]|
|:large_blue_circle:|src/exec/partition/chunks_partitioner.h|0|2|00.00%|[78, 224]|
|:large_blue_circle:|src/exec/pipeline/operator.cpp|3|6|50.00%|[58, 82, 83]|
|:large_blue_circle:|src/exec/spill/spiller.cpp|1|1|100.00%|[]|

File: be/src/connector/lake_connector.cpp (hunk 1 above)
Comment (liuyehcf) [reply be/src/connector/lake_connector.cpp:696]: '_io_statistics_timer' this metric is only to offer hierarchy, and this same time can be represents by `_io_ns_total_timer `, so I remove it here, and change its type from `TIME` to `NONE`

**Review by fzhedu — APPROVED** [review]

**Review by ZiheLiu — APPROVED** [review]

**Review by silverbullet233 — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 25 / 28 (89.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AnalyzeProfileStmt.java|0|1|00.00%|[34]|
|:large_blue_circle:|com/starrocks/qe/scheduler/QueryRuntimeProfile.java|1|2|50.00%|[516]|
|:large_blue_circle:|com/starrocks/common/util/ProfilingExecPlan.java|6|7|85.71%|[398]|
|:large_blue_circle:|com/starrocks/qe/scheduler/dag/FragmentInstanceExecState.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/RuntimeProfileParser.java|11|11|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/RuntimeProfile.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 8 / 136 (05.88%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/bucket_process_operator.h|0|2|00.00%|[32, 63]|
|:large_blue_circle:|src/exec/pipeline/sort/local_partition_topn_sink.cpp|0|3|00.00%|[43, 44, 45]|
|:large_blue_circle:|src/util/runtime_profile.cpp|0|4|00.00%|[1001, 1002, 1003, 1009]|
|:large_blue_circle:|src/connector/lake_connector.cpp|0|52|00.00%|[529, 530, 531, 532, 533, 535, 537, 538, 539, 540, 541, 542, 543, 544, 545, 546, 547, 550, 551, 552, 553, 554, 555, 556, 557, 558, 559, 560, 561, 562, 564, 569, 570, 572, 573, 575, 576, 577, 578, 579, 582, 584, 586, 588, 590, 591, 592, 593, 594, 596, 597, 598]|
|:large_blue_circle:|src/exec/pipeline/scan/olap_chunk_source.cpp|0|41|00.00%|[88, 99, 100, 101, 102, 105, 106, 107, 108, 109, 111, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 140, 143, 333, 502, 506, 510]|
|:large_blue_circle:|src/exec/pipeline/sort/local_partition_topn_context.cpp|0|1|00.00%|[88]|
|:large_blue_circle:|src/exec/partition/partition_hash_variant.cpp|0|17|00.00%|[135, 136, 137, 139, 140, 145, 146, 157, 158, 159, 161, 162, 166, 167, 168, 170, 171]|
|:large_blue_circle:|src/exec/pipeline/sort/local_partition_topn_context.h|0|2|00.00%|[71, 73]|
|:large_blue_circle:|src/exec/partition/chunks_partitioner.h|0|1|00.00%|[78]|
|:large_blue_circle:|src/exec/hdfs_scanner_orc.cpp|0|1|00.00%|[541]|
|:large_blue_circle:|src/exec/pipeline/operator.cpp|3|6|50.00%|[58, 82, 83]|
|:large_blue_circle:|src/connector/hive_connector.cpp|2|3|66.67%|[303]|
|:large_blue_circle:|src/connector/file_connector.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/exec/hdfs_scanner_parquet.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/exec/spill/spiller.cpp|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**